src/App/App.xaml.cs

```diff
@@ -140,6 +136,10 @@ public App(AppOptions appOptions)
                                 new NavigationPage(new RemoveMasterPasswordPage()));
                         });
                     }
+                    else if (message.Command == "passwordlessLoginRequest" || message.Command == "unlocked")
+                    {
+                        Device.BeginInvokeOnMainThread(() => CheckPasswordlessLoginRequestsAsync().FireAndForget());
```

Comment:
I don't think you need the whole method to be invoked on the main thread, do you?. Could you check if you can move that to the specific part where you need it inside `CheckPasswordlessLoginRequestsAsync`? Beware that if you use that inside the async method use `Device.InvokeOnMainThreadAsync` so the inner exceptions bubble up.